packages/py-api/py_api/controllers/hackathon_teams_controller.py

```diff
@@ -0,0 +1,230 @@
+import json
+from typing import Any, Dict, List, Optional, Set
+
+from bson.json_util import dumps
+from bson.objectid import ObjectId
+from fastapi.responses import JSONResponse
+from py_api.database.initialize import t_col
+from py_api.models.hackathon_teams_models import (
+    HackathonTeam,
+    MoveTeamMembers,
+    TeamType,
+    UpdateTeam,
+)
+from py_api.utilities.parsers import filter_none_values
+from pymongo.results import UpdateResult
+
+
+class TeamsController:
+    @classmethod
+    def create_team(
+            cls, team_name: str, user_id: str = "",
+            members_list: Optional[List[str]] = None,
+    ) -> HackathonTeam | None:
+        team_type = TeamType.NORMAL
+
+        if not team_name:
+            # If no team_name is provided during registration, the participant is registering individually.
+            # We create a team of type RANDOM which should be filled with such participants
+            team_type = TeamType.RANDOM
+            team_name = cls.generate_random_team_name()
+
+        team_members = []
+
+        if user_id:
+            team_members = [user_id]
+
+        if members_list:
+            team_members = list(members_list)
+            # Assumes that we are creating a random team when moving members from existing team to a non-existing one
+            team_type = TeamType.RANDOM
```

Comment:
Okay. Not the biggest fan of having so much logic nested in one place, but let's leave it as is. I believe it would've been better if we had a separate method and route for moving participants from one team to another. There's no easy way for us to move a participant from one team to another without creating a new team now.